src/update-feeds/check-feeds.ts

```diff
@@ -0,0 +1,164 @@
+import { go } from '@api3/promise-utils';
+import { ethers } from 'ethers';
+
+import { calculateMedian, checkUpdateConditions } from '../condition-check';
+import { logger } from '../logger';
+import { getStoreDataPoint } from '../signed-data-store';
+import { getState } from '../state';
+import type { ChainId, ProviderName } from '../types';
+import { multiplyBigNumber } from '../utils';
+
+import { getApi3ServerV1 } from './api3-server-v1';
+import type { ReadDapiWithIndexResponse } from './dapi-data-registry';
+import { decodeBeaconValue } from './update-feeds';
+import type { UpdatableDapi } from './update-transactions';
+
+export const checkFeeds = async (
+  batch: ReadDapiWithIndexResponse[],
+  deviationThresholdCoefficient: number,
+  providerName: ProviderName,
+  chainId: ChainId
+): Promise<UpdatableDapi[]> => {
+  const uniqueBeaconIds = [
+    ...new Set(batch.flatMap((item) => item.decodedDataFeed.beacons.flatMap((beacon) => beacon.beaconId))),
+  ];
+  const onChainFeedValues = await callAndParseMulticall(uniqueBeaconIds, providerName, chainId);
+
+  // Merge the latest values into the batch
+  return (
+    batch
+      .map((dapi) => ({
+        ...dapi,
+        decodedDataFeed: {
+          ...dapi.decodedDataFeed,
+          beacons: dapi.decodedDataFeed.beacons.map((beacon) => {
+            const onChainValue = onChainFeedValues.find((onChainValue) => beacon.beaconId === onChainValue.beaconId)
+              ?.onChainValue ?? {
+              value: ethers.BigNumber.from(1),
+              timestamp: ethers.BigNumber.from(1),
+            };
+            const signedData = getStoreDataPoint(beacon.beaconId) ?? {
+              timestamp: '1',
+              airnode: ethers.constants.AddressZero,
+              encodedValue: ethers.constants.HashZero,
+              templateId: ethers.constants.HashZero,
+              signature: ethers.constants.HashZero,
+            };
+            const offChainValue = {
+              timestamp: ethers.BigNumber.from(signedData.timestamp),
+              value: decodeBeaconValue(signedData.encodedValue)!,
+            };
+
+            const localDataIsStale =
```

Comment:
nit: `localDataIsStale` -> `localDataIsValid` because of the future check as well. But good point about that.

Reply:
😆 indeed 🙈